### PR TITLE
Doc: Minor fix to C API documentation

### PR DESCRIPTION
**Pull request by tomerv**

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

## Recognized GitHub username

We couldn't find a [bugs.python.org](https://bugs.python.org/) (b.p.o) account corresponding to the following GitHub usernames:

@tomerv

This might be simply due to a missing "GitHub Name" entry in one's b.p.o account settings. This is necessary for legal reasons before we can look at this contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.




You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Comment by WildCard65** [comment]
@tomerv This pull request requires a bugs.python.org issue number as well as a news file stating the change.
Plus you need to sign the CLA.

**Comment by tomerv** [comment]
@WildCard65 , I submitted the CLA form immediately after getting the automatic message from @the-knights-who-say-ni . Maybe it takes a while to get through?

Regarding bpo issue and news file, did you have a chance to take a look at the change? It's not a lot more than a typo fix. Filing an issue for it seems like a big overhead. Let me know if it's necessary. Same for the news file - I don't think it's necessary to point out every typo fix in the documentation.

**Review by tirkarthi — APPROVED** [review]
LGTM. Thanks.

File: Doc/c-api/init_config.rst [hunk 1]
```diff
@@ -197,12 +197,12 @@ PyPreConfig
...
+   .. c:function:: void PyPreConfig_InitPythonConfig(PyPreConfig *preconfig)
 
       Initialize the preconfiguration with :ref:`Python Configuration
       <init-python-config>`.
 
-   .. c:function:: void PyPreConfig_InitPythonConfig(PyPreConfig *preconfig)
+   .. c:function:: void PyPreConfig_InitIsolatedConfig(PyPreConfig *preconfig)
 
       Initialize the preconfiguration with :ref:`Isolated Configuration
       <init-isolated-conf>`.
```
Comment (vstinner) [inline Doc/c-api/init_config.rst:208]: nitpick: I would prefer to exchange the description but keep the function in the alphabetical order :-)

Comment (tomerv) [reply Doc/c-api/init_config.rst:208]: anti-nitpick: this is consistent with the order for PyConfig functions later in the same file :-)
https://github.com/python/cpython/pull/21198/files#diff-37eacc7b206686c260b26a4fa52ebde4L347

I also like that the more common use case appears first.

**Pull request merged**